platform/security/sts/security-sts-ldapclaimshandler/src/test/java/ddf/security/sts/claimsHandler/RoleClaimsHandlerTest.java

```diff
@@ -14,23 +14,126 @@
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.anyVararg;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Matchers.matches;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.net.URI;
+import java.util.List;
 
 import org.apache.cxf.rt.security.claims.ClaimCollection;
 import org.apache.cxf.sts.claims.ClaimsParameters;
+import org.apache.cxf.sts.claims.ProcessedClaim;
 import org.apache.cxf.sts.claims.ProcessedClaimCollection;
-import org.hamcrest.CoreMatchers;
-import org.junit.Assert;
+import org.apache.karaf.jaas.boot.principal.UserPrincipal;
+import org.forgerock.opendj.ldap.Connection;
+import org.forgerock.opendj.ldap.LDAPConnectionFactory;
+import org.forgerock.opendj.ldap.LdapException;
+import org.forgerock.opendj.ldap.LinkedAttribute;
+import org.forgerock.opendj.ldap.SearchResultReferenceIOException;
+import org.forgerock.opendj.ldap.responses.BindResult;
+import org.forgerock.opendj.ldap.responses.SearchResultEntry;
+import org.forgerock.opendj.ldif.ConnectionEntryReader;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
 
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({LDAPConnectionFactory.class})
 public class RoleClaimsHandlerTest {
+    public static final String USER_CN = "tstark";
+
     @Test
     public void testRetrieveClaimsValuesNullPrincipal() {
         RoleClaimsHandler claimsHandler = new RoleClaimsHandler();
         ClaimsParameters claimsParameters = new ClaimsParameters();
         ClaimCollection claimCollection = new ClaimCollection();
-        ProcessedClaimCollection processedClaims = claimsHandler.retrieveClaimValues(
-                claimCollection, claimsParameters);
+        ProcessedClaimCollection processedClaims =
+                claimsHandler.retrieveClaimValues(claimCollection, claimsParameters);
+        assertThat(processedClaims, is(empty()));
+    }
+
+    @Test
+    public void testRetrieveClaimsValuesNotNullPrincipal()
+            throws LdapException, SearchResultReferenceIOException {
+        BindResult bindResult;
+        ClaimsParameters claimsParameters;
+        Connection connection;
+        ConnectionEntryReader membershipReader, groupNameReader;
+        LDAPConnectionFactory connectionFactory;
+        LinkedAttribute membershipAttribute, groupNameAttribute;
+        ProcessedClaimCollection processedClaims;
+        RoleClaimsHandler claimsHandler;
+        SearchResultEntry membershipSearchResult, groupNameSearchResult;
+        String groupName = "avengers";
+
+        bindResult = mock(BindResult.class);
```

Comment:
Could do the mock inline with the declaration to reduce number of lines for this and the other mocked objects below